src/invoke.rs

```diff
@@ -130,18 +58,35 @@ impl Cmd {
         });
         let storage = Storage::with_recording_footprint(snap);
 
-        let h = Host::with_storage(storage);
+        let mut h = Host::with_storage(storage);
 
-        //TODO: contractID should be user specified
-        let vm = Vm::new(&h, [0; 32].into(), &contents).unwrap();
-        let args = self
+        let args = &self
             .args
             .iter()
-            .map(|a| strval::from_string(&h, a))
-            .collect::<Result<Vec<ScVal>, StrValError>>()?;
-        let res = vm.invoke_function(&h, &self.function, &ScVec(args.try_into()?))?;
-        let res_str = strval::to_string(&h, res);
-        println!("{}", res_str);
+            .map(|a| serde_json::from_str(a))
+            .collect::<Result<Vec<ScVal>, serde_json::Error>>()?;
+
+        if let Some(f) = &self.file {
+            let contents = fs::read(f).unwrap();
+
+            //TODO: contractID should be user specified
```

Comment:
I think this should be resolved by always requiring `--contract-id`, and by essentially having the file uploaded overwrite whatever is already stored at that ID. This type of behavior will have the CLI behave a little more intuitively when someone executes a contract by file, then executes a second contract by file that references the former. In this case the former is also stored and can be executed by the second contract.

Reply:
I'm not sure I understand. This block is only executed if we're executing a WASM blob directly, so the contract won't be stored. The use case of specifying the contractID here is only useful if the contract uses the contractID internally. I don't think we should write the contract to storage using the `invoke` command. That's what `deploy` is for.

Reply:
The invocation will still be operating on the ledger state, so I think it should be possible for a contract to access state of the given contract ID from prior runs. The fact we're disallowing an ID to be provided just means it is always accessing the 0x0 state.

>I don't think we should write the contract to storage using the invoke command. That's what deploy is for.
Having to run deploy manually is inconvenient for local testing. I think the `invoke` command when provided with the file should be a shortcut for deploying that file, to the contract ID, and invoking it. This is essentially what it is doing with the zero ID, but without persisting.

This will make for convenient workflows where a developer can run a command sequence like the below, and where the invocations of the my.wasm and my2.wasm can be associated with distinct contract data that doesn't overlap. It will then also allow my2.wasm to call my.wasm if that's useful.

```
stellar-contract-cli invoke --id 0x1 --file my.wasm --fn go --arg 1
stellar-contract-cli invoke --id 0x1 --file my.wasm --fn go --arg 2
stellar-contract-cli invoke --id 0x2 --file my2.wasm --fn doit
stellar-contract-cli invoke --id 0x1 --file my.wasm --fn go --arg 3
```

The alternative, requiring explicit deploy commands, is less convenient, especially when iterating on a contract, where you might deploy after each run:

```
stellar-contract-cli deploy --id 0x1 --file my.wasm
stellar-contract-cli invoke --id 0x1 --fn go --arg 1
stellar-contract-cli deploy --id 0x1 --file my.wasm
stellar-contract-cli invoke --id 0x1 --fn go --arg 2
stellar-contract-cli deploy --id 0x2 --file my2.wasm
stellar-contract-cli invoke --id 0x2 --fn doit
stellar-contract-cli deploy --id 0x1 --file my.wasm
stellar-contract-cli invoke --id 0x1 --fn go --arg 3
```

Reply:
Ah I see what you're saying. Remove `deploy`, always specify `contractID` so you can access any contract data, and write the contract specified in `invoke` to storage. Yeah I think this is a good idea. I'll make this change now.

Reply:
Done. I kept `deploy` as we discussed, but `invoke` now deploys the contract as well if file is specified.